python/ray/train/huggingface/transformers/_transformers_utils.py

```diff
@@ -213,3 +222,74 @@ def on_train_end(self, args, state, control, **kwargs):
             **self.delayed_report["metrics"],
         }
         self._report()
+
+
+@PublicAPI(stability="alpha")
+class RayTrainReportCallback(TrainerCallback):
+    """A simple callback to report checkpoints and metrics to Ray Tarin.
+
+    This callback overrides the `TrainerCallback.on_save()` method. After
+    a new checkpoint get saved, it fetches the latest metric dictionary
+    from `TrainerState.log_history` and reports it with the latest checkpoint
+    to Ray Train.
+
+    If you want more customized reporting logics, please implement your own
+    callbacks following the Transformers integration user guides.
+
+    Note that users should ensure that the logging, evaluation, and saving frequencies
+    are properly configured to ensure that the monitoring metric is always up-to-date
+    when `transformers.Trainer` saves a checkpoint.
+
+    Suppose the monitoring metric is reported from evaluation stage:
+
+    Some valid configurations:
+        - evaluation_strategy == save_strategy == "epoch"
+        - evaluation_strategy == save_strategy == "steps", save_steps % eval_steps == 0
+
+    Some invalid configurations:
+        - evaluation_strategy != save_strategy
+        - evaluation_strategy == save_strategy == "steps", save_steps % eval_steps != 0
```

Comment:
I think there is a risk this ends up being confusing/restrictive, but I really like the idea of documenting best practices!

Reply:
probably move this to `loading and saving checkpoint` section?